### ISSUE TITLE
Creating OS or datablock images fails

### ISSUE THREAD

**Issue by qwykx** [OpenNebula/addon-linstor#10]
Hello 

I've tried to get OpenNebula 6 to work with linstor. Everything works perfectly fine except creating new OS or generic storage datablocks which fails in Opennebula with error _Error creating datablock_.

The log itself reports:
```bash
Tue Apr 20 08:00:46 2021 [Z0][ImM][I]: Creating disk at  of 10240Mb (format: raw)
Tue Apr 20 08:00:46 2021 [Z0][ReM][D]: Req:5952 UID:0 one.image.allocate result SUCCESS, 107
Tue Apr 20 08:00:46 2021 [Z0][ReM][D]: Req:3728 UID:0 IP:127.0.0.1 one.image.info invoked , 107, false
Tue Apr 20 08:00:46 2021 [Z0][ReM][D]: Req:3728 UID:0 one.image.info result SUCCESS, "<IMAGE><ID>107</ID><..."
Tue Apr 20 08:00:47 2021 [Z0][ReM][D]: Req:9904 UID:0 IP:127.0.0.1 one.datastore.info invoked , 128, false
Tue Apr 20 08:00:47 2021 [Z0][ReM][D]: Req:9904 UID:0 one.datastore.info result SUCCESS, "<DATASTORE><ID>128</..."
Tue Apr 20 08:00:47 2021 [Z0][ReM][D]: Req:7888 UID:0 IP:127.0.0.1 one.imagepool.info invoked , -2, 0, -200, -1
Tue Apr 20 08:00:47 2021 [Z0][ReM][D]: Req:7888 UID:0 one.imagepool.info result SUCCESS, "<IMAGE_POOL><IMAGE><..."
Tue Apr 20 08:00:47 2021 [Z0][ReM][D]: Req:8816 UID:0 IP:127.0.0.1 one.user.info invoked , 0, false
Tue Apr 20 08:00:47 2021 [Z0][ReM][D]: Req:8816 UID:0 one.user.info result SUCCESS, "<USER><ID>0</ID><GID..."
Tue Apr 20 08:00:54 2021 [Z0][ReM][D]: Req:960 UID:0 IP:127.0.0.1 one.imagepool.info invoked , -2, 0, -200, -1
Tue Apr 20 08:00:54 2021 [Z0][ReM][D]: Req:960 UID:0 one.imagepool.info result SUCCESS, "<IMAGE_POOL><IMAGE><..."
Tue Apr 20 08:00:54 2021 [Z0][ReM][D]: Req:8192 UID:0 IP:127.0.0.1 one.user.info invoked , 0, false
Tue Apr 20 08:00:54 2021 [Z0][ReM][D]: Req:8192 UID:0 one.user.info result SUCCESS, "<USER><ID>0</ID><GID..."
Tue Apr 20 08:00:56 2021 [Z0][ReM][D]: Req:3632 UID:0 IP:127.0.0.1 one.zone.raftstatus invoked
Tue Apr 20 08:00:56 2021 [Z0][ReM][D]: Req:3632 UID:0 one.zone.raftstatus result SUCCESS, "<RAFT><SERVER_ID>-1<..."
Tue Apr 20 08:00:56 2021 [Z0][ReM][D]: Req:8912 UID:0 IP:127.0.0.1 one.vmpool.infoextended invoked , -2, -1, -1, -1
Tue Apr 20 08:00:56 2021 [Z0][ReM][D]: Req:8912 UID:0 one.vmpool.infoextended result SUCCESS, "<VM_POOL><VM><ID>374..."
Tue Apr 20 08:00:56 2021 [Z0][ReM][D]: Req:9984 UID:0 IP:127.0.0.1 one.vmpool.infoextended invoked , -2, -1, -1, -1
Tue Apr 20 08:00:56 2021 [Z0][ReM][D]: Req:9984 UID:0 one.vmpool.infoextended result SUCCESS, "<VM_POOL><VM><ID>374..."
Tue Apr 20 08:00:58 2021 [Z0][ReM][D]: Req:1120 UID:1 IP:127.0.0.1 one.documentpool.info invoked , -2, -1, -1, 100
Tue Apr 20 08:00:58 2021 [Z0][ReM][D]: Req:1120 UID:1 one.documentpool.info result SUCCESS, "<DOCUMENT_POOL></DOC..."
Tue Apr 20 08:00:58 2021 [Z0][ReM][D]: Req:9408 UID:1 IP:127.0.0.1 one.vmpool.monitoring invoked , -2, 0
Tue Apr 20 08:00:58 2021 [Z0][ReM][D]: Req:9408 UID:1 one.vmpool.monitoring result SUCCESS, "<MONITORING_DATA><MO..."
Tue Apr 20 08:00:58 2021 [Z0][ReM][D]: Req:9920 UID:1 IP:127.0.0.1 one.vmpool.infoextended invoked , -2, -1, -1, -1, ""
Tue Apr 20 08:00:58 2021 [Z0][ReM][D]: Req:9920 UID:1 one.vmpool.infoextended result SUCCESS, "<VM_POOL><VM><ID>374..."
Tue Apr 20 08:00:58 2021 [Z0][ImM][I]: Command execution failed (exit code: 1): /var/lib/one/remotes/datastore/linstor/mkfs PERTX0RSSVZFUl9BQ1RJT05fREFUQT48SU1BR0U+PElEPjEwNzwvSUQ+PFVJRD4wPC9VSUQ+PEdJRD4wPC9HSUQ+PFVOQU1FPm9uZWFkbWluPC9VTkFNRT48R05BTUU+b25lYWRtaW48L0dOQU1FPjxOQU1FPnRlc3QtT1M8L05BTUU+PFBFUk1JU1NJT05TPjxPV05FUl9VPjE8L09XTkVSX1U+PE9XTkVSX00+MTwvT1dORVJfTT48T1dORVJfQT4wPC9PV05FUl9BPjxHUk9VUF9VPjA8L0dST1VQX1U+PEdST1VQX00+MDwvR1JPVVBfTT48R1JPVVBfQT4wPC9HUk9VUF9BPjxPVEhFUl9VPjA8L09USEVSX1U+PE9USEVSX00+MDwvT1RIRVJfTT48T1RIRVJfQT4wPC9PVEhFUl9BPjwvUEVSTUlTU0lPTlM+PFRZUEU+MDwvVFlQRT48RElTS19UWVBFPjI8L0RJU0tfVFlQRT48UEVSU0lTVEVOVD4xPC9QRVJTSVNURU5UPjxSRUdUSU1FPjE2MTg4OTg0NDY8L1JFR1RJTUU+PFNPVVJDRT48IVtDREFUQVtdXT48L1NPVVJDRT48UEFUSD48IVtDREFUQVtdXT48L1BBVEg+PEZPUk1BVD48IVtDREFUQVtyYXddXT48L0ZPUk1BVD48RlM+PCFbQ0RBVEFbXV0+PC9GUz48U0laRT4xMDI0MDwvU0laRT48U1RBVEU+NDwvU1RBVEU+PFBSRVZfU1RBVEU+MDwvUFJFVl9TVEFURT48UlVOTklOR19WTVM+MDwvUlVOTklOR19WTVM+PENMT05JTkdfT1BTPjA8L0NMT05JTkdfT1BTPjxDTE9OSU5HX0lEPi0xPC9DTE9OSU5HX0lEPjxUQVJHRVRfU05BUFNIT1Q+LTE8L1RBUkdFVF9TTkFQU0hPVD48REFUQVNUT1JFX0lEPjEyODwvREFUQVNUT1JFX0lEPjxEQVRBU1RPUkU+bGluc3Rvcl9kYXRhc3RvcmU8L0RBVEFTVE9SRT48Vk1TPjwvVk1TPjxDTE9ORVM+PC9DTE9ORVM+PEFQUF9DTE9ORVM+PC9BUFBfQ0xPTkVTPjxURU1QTEFURT48REVWX1BSRUZJWD48IVtDREFUQVtzZF1dPjwvREVWX1BSRUZJWD48L1RFTVBMQVRFPjxTTkFQU0hPVFM+PEFMTE9XX09SUEhBTlM+PCFbQ0RBVEFbTk9dXT48L0FMTE9XX09SUEhBTlM+PENVUlJFTlRfQkFTRT48IVtDREFUQVstMV1dPjwvQ1VSUkVOVF9CQVNFPjxORVhUX1NOQVBTSE9UPjwhW0NEQVRBWzBdXT48L05FWFRfU05BUFNIT1Q+PC9TTkFQU0hPVFM+PC9JTUFHRT48REFUQVNUT1JFPjxJRD4xMjg8L0lEPjxVSUQ+MDwvVUlEPjxHSUQ+MDwvR0lEPjxVTkFNRT5vbmVhZG1pbjwvVU5BTUU+PEdOQU1FPm9uZWFkbWluPC9HTkFNRT48TkFNRT5saW5zdG9yX2RhdGFzdG9yZTwvTkFNRT48UEVSTUlTU0lPTlM+PE9XTkVSX1U+MTwvT1dORVJfVT48T1dORVJfTT4xPC9PV05FUl9NPjxPV05FUl9BPjA8L09XTkVSX0E+PEdST1VQX1U+MTwvR1JPVVBfVT48R1JPVVBfTT4wPC9HUk9VUF9NPjxHUk9VUF9BPjA8L0dST1VQX0E+PE9USEVSX1U+MDwvT1RIRVJfVT48T1RIRVJfTT4wPC9PVEhFUl9NPjxPVEhFUl9BPjA8L09USEVSX0E+PC9QRVJNSVNTSU9OUz48RFNfTUFEPjwhW0NEQVRBW2xpbnN0b3JdXT48L0RTX01BRD48VE1fTUFEPjwhW0NEQVRBW2xpbnN0b3JdXT48L1RNX01BRD48QkFTRV9QQVRIPjwhW0NEQVRBWy92YXIvbGliL29uZS8vZGF0YXN0b3Jlcy8xMjhdXT48L0JBU0VfUEFUSD48VFlQRT4wPC9UWVBFPjxESVNLX1RZUEU+MjwvRElTS19UWVBFPjxTVEFURT4wPC9TVEFURT48Q0xVU1RFUlM+PElEPjA8L0lEPjwvQ0xVU1RFUlM+PFRPVEFMX01CPjk1Mzg2ODA8L1RPVEFMX01CPjxGUkVFX01CPjk1MzQ4NjQ8L0ZSRUVfTUI+PFVTRURfTUI+MzgxNjwvVVNFRF9NQj48SU1BR0VTPjxJRD4xMDI8L0lEPjxJRD4xMDM8L0lEPjxJRD4xMDQ8L0lEPjwvSU1BR0VTPjxURU1QTEFURT48QUxMT1dfT1JQSEFOUz48IVtDREFUQVt5ZXNdXT48L0FMTE9XX09SUEhBTlM+PEJSSURHRV9MSVNUPjwhW0NEQVRBW3N0bzA1MDEgc3RvMDUwMl1dPjwvQlJJREdFX0xJU1Q+PENMT05FX1RBUkdFVD48IVtDREFUQVtTRUxGXV0+PC9DTE9ORV9UQVJHRVQ+PENMT05FX1RBUkdFVF9TSEFSRUQ+PCFbQ0RBVEFbU0VMRl1dPjwvQ0xPTkVfVEFSR0VUX1NIQVJFRD48Q0xPTkVfVEFSR0VUX1NTSD48IVtDREFUQVtTRUxGXV0+PC9DTE9ORV9UQVJHRVRfU1NIPjxDT01QQVRJQkxFX1NZU19EUz48IVtDREFUQVsxMjldXT48L0NPTVBBVElCTEVfU1lTX0RTPjxESVNLX1RZUEU+PCFbQ0RBVEFbQkxPQ0tdXT48L0RJU0tfVFlQRT48RElTS19UWVBFX1NIQVJFRD48IVtDREFUQVtCTE9DS11dPjwvRElTS19UWVBFX1NIQVJFRD48RElTS19UWVBFX1NTSD48IVtDREFUQVtCTE9DS11dPjwvRElTS19UWVBFX1NTSD48RFNfTUFEPjwhW0NEQVRBW2xpbnN0b3JdXT48L0RTX01BRD48TElOU1RPUl9DT05UUk9MTEVSUz48IVtDREFUQVsxMC4xOTkuOTkuMTk6MzM3MF1dPjwvTElOU1RPUl9DT05UUk9MTEVSUz48TElOU1RPUl9SRVNPVVJDRV9HUk9VUD48IVtDREFUQVtvbmVncm91cF1dPjwvTElOU1RPUl9SRVNPVVJDRV9HUk9VUD48TE5fVEFSR0VUPjwhW0NEQVRBW05PTkVdXT48L0xOX1RBUkdFVD48TE5fVEFSR0VUX1NIQVJFRD48IVtDREFUQVtOT05FXV0+PC9MTl9UQVJHRVRfU0hBUkVEPjxMTl9UQVJHRVRfU1NIPjwhW0NEQVRBW05PTkVdXT48L0xOX1RBUkdFVF9TU0g+PFJFU1RSSUNURURfRElSUz48IVtDREFUQVsvXV0+PC9SRVNUUklDVEVEX0RJUlM+PFNBRkVfRElSUz48IVtDREFUQVsvdmFyL3RtcF1dPjwvU0FGRV9ESVJTPjxUTV9NQUQ+PCFbQ0RBVEFbbGluc3Rvcl1dPjwvVE1fTUFEPjxUTV9NQURfU1lTVEVNPjwhW0NEQVRBW3NzaCxzaGFyZWRdXT48L1RNX01BRF9TWVNURU0+PFRZUEU+PCFbQ0RBVEFbSU1BR0VfRFNdXT48L1RZUEU+PC9URU1QTEFURT48L0RBVEFTVE9SRT48L0RTX0RSSVZFUl9BQ1RJT05fREFUQT4= 107
Tue Apr 20 08:00:58 2021 [Z0][ImM][E]: Error creating datablock
```

Unfortunately I don't see what should be wrong looking at the _mkfs_ script itself :/

Cheers

**Comment by rp-** [comment]
is there more info in the journal?
`journalctl -u opennebula`

**Comment by qwykx** [comment]
Here is the corresponding error from _journalctl_

```bash
Apr 20 08:00:47 HVS0501 /mkfs[8187]: INFO Entering datastore mkfs.
Apr 20 08:00:47 HVS0501 /mkfs[8187]: INFO running shell command: bash -c source /var/lib/one/remotes//datastore/libfs.sh && set_up_datastore /var/lib/one//datastores/128 / /var/tmp
Apr 20 08:00:47 HVS0501 /mkfs[8187]: INFO Creation of a new resource: OpenNebula-Image-107
Apr 20 08:00:47 HVS0501 /mkfs[8187]: INFO Deploying resource 'OpenNebula-Image-107' using resource group 'onegroup', prefer node: None
Apr 20 08:00:55 HVS0501 /mkfs[8187]: INFO ssh 'sto0501' cmd: set -e ; export PATH=/usr/sbin:/sbin:$PATH ; source /var/lib/one/remotes/etc/datastore/datastore.conf ; FS_OPTS=$(eval $(echo "echo \$FS_OPTS_raw")) ; $(mkfs_command "/dev/drbd1018" "raw" "10240" "$SUPPORTED_FS" "" "$FS_OPTS")
Apr 20 08:00:56 HVS0501 /mkfs[8187]: ERROR command ['bash', '-c', 'source /var/lib/one/remotes//scripts_common.sh && ssh_exec_and_log "sto0501" "set -e ; export PATH=/usr/sbin:/sbin:$PATH ; source /var/lib/one/remotes/etc/datastore/datastore.conf ; FS_OPTS=$(eval $(echo "echo \\$FS_OPTS_raw")) ; $(mkfs_command "/dev/drbd1018" "raw" "10240" "$SUPPORTED_FS" "" "$FS_OPTS")" "Error mkfs_command for resource OpenNebula-Image-107(OpenNebula-Image-107), on sto0501"'] failed: b'ERROR: bash: Command "set -e ; export PATH=/usr/sbin:/sbin:/bin:/sbin:/usr/bin:/usr/lib/one/sh/override:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin ; source /var/lib/one/remotes/etc/datastore/datastore.conf ; FS_OPTS= ; qemu-img create -f raw /dev/drbd1018 10240M" failed: bash: line 3: /var/lib/one/remotes/etc/datastore/datastore.conf: No such file or directory\nERROR MESSAGE --8<------\nError mkfs_command for resource OpenNebula-Image-107(OpenNebula-Image-107), on sto0501\nERROR MESSAGE ------>8--\n'

```

**Comment by rp-** [comment]
From the error it looks like you don't have the file `/var/lib/one/remotes/etc/datastore/datastore.conf` on your host `sto0501`, my test setup has this.
Can you show also your datastore settings: `onedatastore show <ds_ID>`

**Comment by qwykx** [comment]
The server `sto0501` is a dedicated storage server and is not a compute node for opennebula. But it might be that I got the instructions wrong from linstor and have therefore configured it wrong somehow.

```bash
ALLOW_ORPHANS="yes"
BRIDGE_LIST="sto0501 sto0502"
CLONE_TARGET="SELF"
CLONE_TARGET_SHARED="SELF"
CLONE_TARGET_SSH="SELF"
COMPATIBLE_SYS_DS="129"
DISK_TYPE="BLOCK"
DISK_TYPE_SHARED="BLOCK"
DISK_TYPE_SSH="BLOCK"
DS_MAD="linstor"
LINSTOR_CONTROLLERS="10.199.99.19:3370"
LINSTOR_RESOURCE_GROUP="onegroup"
LN_TARGET="NONE"
LN_TARGET_SHARED="NONE"
LN_TARGET_SSH="NONE"
RESTRICTED_DIRS="/"
SAFE_DIRS="/var/tmp"
TM_MAD="linstor"
TM_MAD_SYSTEM="ssh,shared"
TYPE="IMAGE_DS"
```

**Comment by rp-** [comment]
I checked another fresh test setup, there the file is also missing. :shrug: 

As a quick fix you can edit: `/var/lib/one/remotes/datastore/linstor/mkfs`
and remove line 84: `source {remotes}etc/datastore/datastore.conf ; \`

**Comment by qwykx** [comment]
Working now 👍🏼

**Comment by rp-** [comment]
I released a new version with a correct fix: v1.1.1
